external-crates/move/crates/move-package-alt/src/graph/rename_from.rs

```diff
@@ -172,4 +190,84 @@ mod tests {
         assert_snapshot!(scenario.graph_for("root").await.check_rename_from().unwrap_err().to_string(), @"TODO");
         assert_snapshot!(scenario.graph_for("a").await.check_rename_from().unwrap_err().to_string(), @"TODO");
     }
+
+    #[test(tokio::test)]
+    async fn test_modern_using_legacy_framework() {
+        let scenario = TestPackageGraph::new(["root"])
+            .add_package("std", |pkg| pkg.set_legacy().set_legacy_name("MoveStdLib"))
+            .add_package("sui", |pkg| pkg.set_legacy().set_legacy_name("Sui"))
+            .add_package("sui_system", |pkg| {
+                pkg.set_legacy().set_legacy_name("SuiSystem")
+            })
+            .add_deps([("root", "std")])
+            .add_dep("root", "sui", |dep| dep.name("my_sui").rename_from("sui"))
+            .add_dep("root", "sui_system", |dep| {
+                dep.name("my_sui_system").rename_from("sui_system")
+            })
+            .add_dep("sui", "std", |dep| dep.name("MoveStdLib"))
+            // legacy -> legacy case (SuiSystem -> MoveStdLib (std)
+            .add_dep("sui_system", "std", |dep| dep.name("MoveStdLib"))
+            .build();
+
+        scenario
+            .graph_for("root")
+            .await
+            .check_rename_from()
+            .unwrap();
+
+        scenario
+            .graph_for("sui_system")
+            .await
+            .check_rename_from()
+            .unwrap();
+    }
+
+    #[test(tokio::test)]
+    async fn test_modern_to_legacy_not_allowed_behaviours() {
+        let scenario = TestPackageGraph::new(["root", "foo", "bar", "baz"])
+            .add_package("legacy", |pkg| pkg.set_legacy().set_legacy_name("Legacy"))
+            .add_package("sui", |pkg| pkg.set_legacy().set_legacy_name("Sui"))
+            .add_package("std", |pkg| pkg.set_legacy().set_legacy_name("MoveStdLib"))
+            .add_package("malformed", |pkg| {
+                pkg.set_legacy().set_legacy_name("weird-input")
+            })
+            // 1. (FAIL) Cannot use the legacy name in the left side assignment
+            .add_dep("root", "sui", |dep| dep.name("Sui"))
+            // 2. (OK) Can use the "modern" name in the rename-from (even if we name it as the legacy name)
+            .add_dep("foo", "std", |dep| {
+                dep.name("MoveStdLib").rename_from("std")
+            })
+            // 3. (FAIL) Cannot use the legacy name in the rename-from
+            .add_dep("bar", "std", |dep| {
+                dep.name("foo").rename_from("MoveStdLib")
+            })
+            // 4. (OK) Legacy packages CAN use legacy names freely!
+            .add_dep("legacy", "std", |dep| dep.name("MoveStdLib"))
```

Comment:
two other cases:
5. `.add_dep("legacy2", "std", |dep| dep.name("AnotherThing"))` - maybe should be disallowed (since the legacy system would require `MoveStdlib = { ... }`)
6. `.add_dep("legacy3", "malformed", |dep| dep.name("weird-thing"))` - should work I think because the legacy system would allow it